generic/src/main/scala/io/finch/generic/package.scala

```diff
@@ -0,0 +1,8 @@
+package io.finch
+
+package object generic {
+  /**
+   * Generically derive a very basic instance of [[Endpoint]] for a given type `A`.
+   */
+  def deriveEndpointFor[A]: GenericDerivation[A] = new GenericDerivation[A]
```

Comment:
Yeah, I can see why you might like it and yet I think it doesn't really add much readability. Also, I learn from Circe that has `deriveDecoder` and `deriveEncoder`.